sky/backends/backend_utils.py

```diff
@@ -763,7 +763,14 @@ def write_cluster_config(
                     keys=('allowed_contexts',),
                     default_value=None)
             if allowed_contexts is None:
-                excluded_clouds.add(cloud)
+                # Exclude both Kubernetes and SSH explicitly since:
+                # 1. isinstance(cloud, clouds.Kubernetes) matches both (SSH
+                #    inherits from Kubernetes)
+                # 2. Both share the same get_credential_file_mounts() which
+                #    returns the kubeconfig. So if we don't exclude both, the
+                #    unexcluded one will upload the kubeconfig.
+                excluded_clouds.add(clouds.Kubernetes())
+                excluded_clouds.add(clouds.SSH())
```

Comment:
We may want to add a TODO to better handle this. This feels quite hacky. 

In the future, we should have the SSH to have its own `get_credential_file_mounts`

Reply:
Agreed. Left a note. The right long-term fix is to have SSH Node Pools use it's own kubeconfig. In the interim, SSH's `get_credential_file_mounts` can filter contexts starting with `ssh-` and create a temp kubeconfig to upload.